diff --git a/b/src/drone b/b/src/drone
new file mode 120000
index 0000000000..2b76d29a42
--- /dev/null
+++ b/b/src/drone
@@ -0,0 +1 @@
+../../drone
\ No newline at end of file
diff --git a/b/src/report b/b/src/report
new file mode 120000
index 0000000000..4bc1423586
--- /dev/null
+++ b/b/src/report
@@ -0,0 +1 @@
+../../report
\ No newline at end of file
diff --git a/src/drone b/src/drone
new file mode 120000
index 0000000000..2b76d29a42
--- /dev/null
+++ b/src/drone
@@ -0,0 +1 @@
+../../drone
\ No newline at end of file
diff --git a/src/lte/helper/lte-helper.h b/src/lte/helper/lte-helper.h
index 2731be7abc..fd5ce6fbc7 100644
--- a/src/lte/helper/lte-helper.h
+++ b/src/lte/helper/lte-helper.h
@@ -102,6 +102,8 @@ class SpectrumPropagationLossModel;
 class LteHelper : public Object
 {
 public:
+  friend class LteSetupHelper;
+
   LteHelper (void);
   virtual ~LteHelper (void);
diff --git a/src/report b/src/report
new file mode 120000
index 0000000000..4bc1423586
--- /dev/null
+++ b/src/report
@@ -0,0 +1 @@
+../../report
\ No newline at end of file
diff --git a/src/wifi/helper/wifi-helper.h b/src/wifi/helper/wifi-helper.h
index b15888e2f3..d8f7cf1b1f 100644
--- a/src/wifi/helper/wifi-helper.h
+++ b/src/wifi/helper/wifi-helper.h
@@ -326,6 +326,8 @@ private:
 class WifiHelper
 {
 public:
+
+  friend class WifiMacFactoryHelper;
   virtual ~WifiHelper ();

   /**
diff --git a/src/wifi/helper/yans-wifi-helper.h b/src/wifi/helper/yans-wifi-helper.h
index 61193c80ba..5d4324bc77 100644
--- a/src/wifi/helper/yans-wifi-helper.h
+++ b/src/wifi/helper/yans-wifi-helper.h
@@ -36,6 +36,8 @@ namespace ns3 {
 class YansWifiChannelHelper
 {
 public:
+  friend class WifiPhyFactoryHelper;
+
   /**
    * Create a channel helper without any parameter set. The user must set
    * them all to be able to call Create later.
diff --git a/src/mobility/helper/mobility-helper.h b/src/mobility/helper/mobility-helper.h
index 0ea23a9f3..dcee375be 100644
--- a/src/mobility/helper/mobility-helper.h
+++ b/src/mobility/helper/mobility-helper.h
@@ -42,6 +42,8 @@ class MobilityModel;
 class MobilityHelper
 {
 public:
+  friend class MobilityFactoryHelper;
+
   /**
    * Construct a Mobility Helper which is used to make life easier when working
    * with mobility models.
--
2.25.1
