diff --git a/src/lte/helper/lte-helper.h b/src/lte/helper/lte-helper.h
index 0bb8b1a6e..36d74a8bf 100644
--- a/src/lte/helper/lte-helper.h
+++ b/src/lte/helper/lte-helper.h
@@ -102,6 +102,8 @@ class SpectrumPropagationLossModel;
 class LteHelper : public Object
 {
   public:
+    friend class LteSetupHelper;
+
     LteHelper();
     ~LteHelper() override;
 
diff --git a/src/lte/helper/no-backhaul-epc-helper.cc b/src/lte/helper/no-backhaul-epc-helper.cc
index 98ce23bab..92dd4a1ad 100644
--- a/src/lte/helper/no-backhaul-epc-helper.cc
+++ b/src/lte/helper/no-backhaul-epc-helper.cc
@@ -331,8 +331,11 @@ NoBackhaulEpcHelper::AddEnb(Ptr<Node> enb,
     int retval;
 
     // add an IPv4 stack to the previously created eNB
-    InternetStackHelper internet;
-    internet.Install(enb);
+    if (enb->GetObject<Ipv4>() == nullptr)
+    {
+        InternetStackHelper internet;
+        internet.Install(enb);
+    }
     NS_LOG_LOGIC("number of Ipv4 ifaces of the eNB after node creation: "
                  << enb->GetObject<Ipv4>()->GetNInterfaces());
 
@@ -370,10 +373,10 @@ NoBackhaulEpcHelper::AddEnb(Ptr<Node> enb,
     Ptr<EpcEnbApplication> enbApp =
         CreateObject<EpcEnbApplication>(enbLteSocket, enbLteSocket6, cellIds.at(0));
     enb->AddApplication(enbApp);
-    NS_ASSERT(enb->GetNApplications() == 1);
-    NS_ASSERT_MSG(enb->GetApplication(0)->GetObject<EpcEnbApplication>(),
-                  "cannot retrieve EpcEnbApplication");
-    NS_LOG_LOGIC("enb: " << enb << ", enb->GetApplication (0): " << enb->GetApplication(0));
+    // NS_ASSERT(enb->GetNApplications() == 1);
+    // NS_ASSERT_MSG(enb->GetApplication(0)->GetObject<EpcEnbApplication>(),
+    //               "cannot retrieve EpcEnbApplication");
+    // NS_LOG_LOGIC("enb: " << enb << ", enb->GetApplication (0): " << enb->GetApplication(0));
 
     NS_LOG_INFO("Create EpcX2 entity");
     Ptr<EpcX2> x2 = CreateObject<EpcX2>();
