diff --git a/src/core/model/log.h b/src/core/model/log.h
index 135c9efad..e2317d32b 100644
--- a/src/core/model/log.h
+++ b/src/core/model/log.h
@@ -112,7 +112,7 @@ enum LogLevel
     LOG_LOGIC = 0x00000020,       //!< Control flow tracing within functions.
     LOG_LEVEL_LOGIC = 0x0000003f, //!< LOG_LOGIC and above.
 
-    LOG_ALL = 0x0fffffff,    //!< Print everything.
+    LOG_ALL = 0xffffffff,    //!< Print everything.
     LOG_LEVEL_ALL = LOG_ALL, //!< Print everything.
 
     LOG_PREFIX_FUNC = 0x80000000,  //!< Prefix all trace prints with function.
